lib/ansible/module_utils/basic.py

```diff
@@ -814,11 +830,15 @@ def __init__(self, argument_spec, bypass_checks=False, no_log=False,
         self._clean = {}
 
         self.aliases = {}
-        self._legal_inputs = ['_ansible_check_mode', '_ansible_no_log', '_ansible_debug', '_ansible_diff', '_ansible_verbosity',
-                              '_ansible_selinux_special_fs', '_ansible_module_name', '_ansible_version', '_ansible_syslog_facility',
-                              '_ansible_socket', '_ansible_shell_executable']
+        self._legal_inputs = ['_ansible_%s' % k for k in PASS_VARS]
         self._options_context = list()
 
+        # set tempdir to remote tmp
+        self.tempdir = os.environ.get('ANSIBLE_REMOTE_TEMP', None)
```

Comment:
We should do this via legal_inputs.  Yes, that will duplicate information that we had to pass to ansiballz via the environment but it keeps the handling of generic information from the controller to a single idiom.

Reply:
"[...] handling of generic information from the controller in AnsibleModule."

Reply:
Decided abadger will change this so that new-style modules get temporary from an _ansible parameter instead of from the environment.